pydantic/_internal/_generate_schema.py

```diff
@@ -978,6 +965,9 @@ def _get_args_resolving_forward_refs(self, obj: Any, required: bool = False) ->
         if args:
             if isinstance(obj, GenericAlias):
                 # PEP 585 generic aliases don't convert args to ForwardRefs, unlike `typing.List/Dict` etc.
+                # This was fixed in https://github.com/python/cpython/pull/30900 (Python 3.11).
+                # TODO: this shouldn't be necessary (probably even this `_get_args_resolving_forward_refs()` function)
+                # once we drop support for Python 3.10 *or* if we implement our own `typing._eval_type()` implementation.
```

Comment:
Should we update the `isinstance` to have a condition on `sys.version_info < (3, 11)` (or even just make the whole function just return early)?